custom_components/owlintuition/sensor.py

```diff
@@ -260,57 +268,61 @@ def __init__(self, owldata, sensor_name, sensor_type, phase=0):
-            ATTR_LAST_UPDATE: None,
-            ATTR_LAST_RESET: None,
-            ATTR_DEVICE_CLASS: SENSOR_TYPES[sensor_type][4],
-            ATTR_STATE_CLASS: STATE_CLASS_TOTAL_INCREASING if SENSOR_TYPES[sensor_type][4] == c.DEVICE_CLASS_ENERGY \
-                              else (STATE_CLASS_MEASUREMENT if SENSOR_TYPES[sensor_type][4] else None),
-            c.ATTR_ATTRIBUTION: 'Powered by OWL Intuition'
-            }
+        self._attr_attribution = POWERED_BY
+        self._attr_device_class = SENSOR_TYPES[self._sensor_type][4]
+        self._attr_native_unit_of_measurement = SENSOR_TYPES[self._sensor_type][1]
+        self._attr_state_class = SENSOR_TYPES[self._sensor_type][5]
 
     @property
-    def name(self):
-        """Return the name of this sensor."""
+    def _attr_name(self):
+        """Return the current name for this sensor."""
         return self._name
 
+    # @property
+    # def _attr_device_class(self):
+    #     """Return the device class of the sensor."""
+    #     return SENSOR_TYPES[self._sensor_type][4]
+
+    # @property
+    # def _attr_native_unit_of_measurement(self):
+    #     """Return the unit of measurement of this entity."""
+    #     return SENSOR_TYPES[self._sensor_type][1]
+
     @property
-    def state(self):
+    def _attr_native_value(self):
         """Return the current value for this sensor."""
         return self._state
 
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity."""
-        return SENSOR_TYPES[self._sensor_type][1]
+    # @property
+    # def _attr_state_class(self):
+    #     """Return the state class of the sensor."""
+    #     return SENSOR_TYPES[self._sensor_type][5]
 
     @property
     def icon(self):
         """Return the icon for this entity."""
         return SENSOR_TYPES[self._sensor_type][2]
 
-    @property
-    def device_state_attributes(self):
-        """Return the state attributes of the sensor."""
-        return self._attrs
-
     def update(self):
         """Retrieve the latest value for this sensor."""
         self._owldata.update()
         xml = self._owldata.get(self._owl_class)
         if xml is None:
             return
         xml_ver = xml.attrib.get('ver')
-        self._attrs[ATTR_LAST_UPDATE] = int(xml.find('timestamp').text)
-        if self._attrs[ATTR_DEVICE_CLASS] == c.DEVICE_CLASS_ENERGY:
-            self._attrs[ATTR_LAST_RESET] = datetime.today().replace(hour=0, minute=0, second=0, microsecond=0)
+        # This never appears to have been setable: https://github.com/home-assistant/architecture/issues/190
+        # self._attrs[ATTR_LAST_UPDATE] = int(xml.find('timestamp').text)
+
+        # Setting last_reset for entities with state_class other than 'total' is not supported.
+        # if self._attr_device_class == SensorDeviceClass.ENERGY:
+        #     self._attr_last_reset = datetime.today().replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
Interesting observation... do you mean the reset logic has now improved such that HA core auto-detects when a given energy-based (and supposedly always growing) entity has a value that get reset?
I remember this was added at the beginning to precisely signal the moment in time when the energy-based values are reset by the integration.

Reply:
@glpatcern There have been changes around how the long term statistics are handled including reset, see: https://developers.home-assistant.io/docs/core/entity/sensor/#entities-representing-a-total-amount

You can either use the state class `TOTAL` and manually control reset with `last_reset` or with the `TOTAL_INCREASING` class "a decreasing value is interpreted as the start of a new meter cycle or the replacement of the meter".

If you attempt to use `last_reset` with `TOTAL_INCREASING` and have HA logging level set to warning, then you get a message logged informing you that it's not supported.

Per: 
```
# This should raise in Home Assistant Core 2022.5
                _LOGGER.warning(
                    "Entity %s (%s) with state_class %s has set last_reset. Setting "
                    "last_reset for entities with state_class other than 'total' is "
                    "not supported. "
                    "Please update your configuration if state_class is manually "
                    "configured, otherwise %s",
                    self.entity_id,
                    type(self),
                    self.state_class,
                    report_issue,
                )

```